src/Microsoft.AspNet.Diagnostics/ErrorPageMiddleware.cs

```diff
@@ -182,31 +188,103 @@ private StackFrame StackFrame(Chunk line)
 
             int lineNumber = line.ToInt32();
 
-            return string.IsNullOrEmpty(file)
-                ? LoadFrame(string.IsNullOrEmpty(function) ? line.ToString() : function, string.Empty, 0)
-                : LoadFrame(function, file, lineNumber);
+            if (string.IsNullOrEmpty(file))
+            {
+                return GetStackFrame(
+                    string.IsNullOrEmpty(function) ? line.ToString() : function,
```

Comment:
I believe its to handle the case where the stack trace contains some text like `--End of stack trace...` 
https://cloud.githubusercontent.com/assets/6559784/8710577/94596bb8-2aff-11e5-853c-3c4c4ea2afa6.png